### ISSUE TITLE
FAKE bug - can't access denied during routine NuGet packaging and cleanup

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/Akka.Persistence.PostgreSql#2]
[05:21:02][Step 1/1] [WARNING][6/26/2015 5:21:01 AM][Thread 0004][akka://PostgreSqlSnapshotStoreSpec/user] DeadLetter from [akka://PostgreSqlSnapshotStoreSpec/user] to [akka://PostgreSqlSnapshotStoreSpec/user]: <Received dead system message: <DeathWatchNotification>: [akka://PostgreSqlSnapshotStoreSpec/user/testActor76], ExistenceConfirmed=True, AddressTerminated=False>
    [05:21:02][Step 1/1] Finished Target: RunTests
    [05:21:02][Step 1/1] Starting Target: Nuget (==> BuildRelease)
    [05:21:02][Step 1/1] Target: Nuget
    [05:21:02][Step 1/1] Creating nuget packages for D:\BuildAgent\work\29e799f160213a61\src\Akka.Persistence.PostgreSql\Akka.Persistence.PostgreSql.nuspec
    [05:21:02][Step 1/1] Creating D:\BuildAgent\work\29e799f160213a61\bin\build
    [05:21:02][Step 1/1] Creating D:\BuildAgent\work\29e799f160213a61\bin\build\src\
    [05:21:02][Step 1/1] D:\BuildAgent\work\29e799f160213a61\bin\build\src\obj does not exist.
    [05:21:02][Step 1/1] D:\BuildAgent\work\29e799f160213a61\bin\build\src\bin does not exist.
    [05:21:02][Step 1/1] Task: NuGet D:\BuildAgent\work\29e799f160213a61\src\Akka.Persistence.PostgreSql\Akka.Persistence.PostgreSql.nuspec
    [05:21:02][Step 1/1] Creating .nuspec file at D:\BuildAgent\work\29e799f160213a61\bin\build\Akka.Persistence.PostgreSql.1.0.4.1-beta.nuspec
    [05:21:02][Step 1/1] Created nuspec file D:\BuildAgent\work\29e799f160213a61\bin\build\Akka.Persistence.PostgreSql.1.0.4.1-beta.nuspec
    [05:21:02][Step 1/1] D:\BuildAgent\work\29e799f160213a61\src\.nuget\NuGet.exe pack -Symbols -Version 1.0.4.1-beta -OutputDirectory "D:\BuildAgent\work\29e799f160213a61\bin\nuget" "Akka.Persistence.PostgreSql.1.0.4.1-beta.nuspec"-Properties Configuration="Release"
    [05:21:03][Step 1/1] Deleting D:\BuildAgent\work\29e799f160213a61\bin\build\Akka.Persistence.PostgreSql.1.0.4.1-beta.nuspec
    [05:21:03][Step 1/1] Deleting D:\BuildAgent\work\29e799f160213a61\bin\build
    [05:21:03][Step 1/1] Task failed with The directory is not empty.
    [05:21:03][Step 1/1] 
    [05:21:03][Step 1/1] Retry.
    [05:21:03][Step 1/1] Task failed with Access to the path 'D:\BuildAgent\work\29e799f160213a61\bin\build\lib\net45' is denied.
    [05:21:03][Step 1/1] Retry.
    [05:21:03][Step 1/1] Task failed with Access to the path 'D:\BuildAgent\work\29e799f160213a61\bin\build\lib\net45' is denied.
    [05:21:03][Step 1/1] Retry.
    [05:21:04][Step 1/1] Running build failed.
    [05:21:04][Step 1/1] ##teamcity[buildStatus status='FAILURE' text='System.UnauthorizedAccessException: Access to the path |'D:\BuildAgent\work\29e799f160213a61\bin\build\lib\net45|' is denied.   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)   at System.IO.FileSystemEnumerableIterator`1.AddSearchableDirsToStack(SearchData localSearchData)   at System.IO.FileSystemEnumerableIterator`1.MoveNext()   at Microsoft.FSharp.Core.CompilerServices.RuntimeHelpers.takeInner@667|[T,TResult|](ConcatEnumerator`2 x, Unit unitVar0)   at Microsoft.FSharp.Collections.SeqModule.ToList|[T|](IEnumerable`1 source)   at Fake.Globbing.buildPaths(FSharpList`1 acc, FSharpList`1 input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 47   at Fake.Globbing.search(String baseDir, String input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 81   at Fake.FileSystem.System-Collections-Generic-IEnumerable-1-GetEnumerator@39-2.GenerateNext(IEnumerable`1& next) in C:\code\fake\src\app\FakeLib\Globbing\FileSystem.fs:line 40   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl()   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System-Collections-IEnumerator-MoveNext()   at Microsoft.FSharp.Collections.IEnumerator.next@185|[T|](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)   at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()   at Microsoft.FSharp.Collections.IEnumerator.next@185|[T|](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)   at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()   at Microsoft.FSharp.Collections.SeqModule.Iterate|[T|](FSharpFunc`2 action, IEnumerable`1 source)   at Fake.FileHelper.DeleteDir(String path) in C:\code\fake\src\app\FakeLib\FileHelper.fs:line 46   at FSI_0001.Build.del@170-1.Invoke(b _arg2) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 170   at Fake.TaskRunnerHelper.runWithRetries|[a|](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32   at Fake.TaskRunnerHelper.runWithRetries|[a|](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32   at Fake.TaskRunnerHelper.runWithRetries|[a|](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32   at FSI_0001.Build.removeDir@168(String dir) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 172   at FSI_0001.Build.createNugetPackages|[a|](a _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 227   at FSI_0001.Build.clo@273-15.Invoke(Unit _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 274   at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in C:\code\fake\src\app\FakeLib\TargetHelper.fs:line 411']
    [05:21:04][Step 1/1] Error:
    [05:21:04][Step 1/1] System.UnauthorizedAccessException: Access to the path 'D:\BuildAgent\work\29e799f160213a61\bin\build\lib\net45' is denied.
    [05:21:04][Step 1/1]at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
    [05:21:04][Step 1/1]at System.IO.FileSystemEnumerableIterator`1.AddSearchableDirsToStack(SearchData localSearchData)
    [05:21:04][Step 1/1]at System.IO.FileSystemEnumerableIterator`1.MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.RuntimeHelpers.takeInner@667[T,TResult](ConcatEnumerator`2 x, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
    [05:21:04][Step 1/1]at Fake.Globbing.buildPaths(FSharpList`1 acc, FSharpList`1 input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 47
    [05:21:04][Step 1/1]at Fake.Globbing.search(String baseDir, String input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 81
    [05:21:04][Step 1/1]at Fake.FileSystem.System-Collections-Generic-IEnumerable-1-GetEnumerator@39-2.GenerateNext(IEnumerable`1& next) in C:\code\fake\src\app\FakeLib\Globbing\FileSystem.fs:line 40
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.next@185[T](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.next@185[T](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.SeqModule.Iterate[T](FSharpFunc`2 action, IEnumerable`1 source)
    [05:21:04][Step 1/1]at Fake.FileHelper.DeleteDir(String path) in C:\code\fake\src\app\FakeLib\FileHelper.fs:line 46
    [05:21:04][Step 1/1]at FSI_0001.Build.del@170-1.Invoke(b _arg2) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 170
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at FSI_0001.Build.removeDir@168(String dir) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 172
    [05:21:04][Step 1/1]at FSI_0001.Build.createNugetPackages[a](a _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 227
    [05:21:04][Step 1/1]at FSI_0001.Build.clo@273-15.Invoke(Unit _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 274
    [05:21:04][Step 1/1]at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in C:\code\fake\src\app\FakeLib\TargetHelper.fs:line 411
    [05:21:04][Step 1/1] 
    [05:21:04][Step 1/1] ---------------------------------------------------------------------
    [05:21:04][Step 1/1] Build Time Report
    [05:21:04][Step 1/1] ---------------------------------------------------------------------
    [05:21:04][Step 1/1] TargetDuration
    [05:21:04][Step 1/1] --------------
    [05:21:04][Step 1/1] Clean 00:00:00.0014046
    [05:21:04][Step 1/1] AssemblyInfo  00:00:00.0130088
    [05:21:04][Step 1/1] RestorePackages   00:00:03.9659438
    [05:21:04][Step 1/1] Build 00:00:01.6350134
    [05:21:04][Step 1/1] CopyOutput00:00:00.0212241
    [05:21:04][Step 1/1] CleanNuget00:00:00.0010638
    [05:21:04][Step 1/1] BuildRelease  00:00:00.0000556
    [05:21:04][Step 1/1] CleanTests00:00:00.0001280
    [05:21:04][Step 1/1] RunTests  00:00:26.9601007
    [05:21:04][Step 1/1] Total:00:00:34.0761731
    [05:21:04][Step 1/1] ---------------------------------------------------------------------
    [05:21:04][Step 1/1] Status:   Failure
    [05:21:04][Step 1/1]   1) System.UnauthorizedAccessException: Access to the path 'D:\BuildAgent\work\29e799f160213a61\bin\build\lib\net45' is denied.
    [05:21:04][Step 1/1] ---------------------------------------------------------------------
    [05:21:04][Step 1/1]at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
    [05:21:04][Step 1/1]at System.IO.FileSystemEnumerableIterator`1.AddSearchableDirsToStack(SearchData localSearchData)
    [05:21:04][Step 1/1]at System.IO.FileSystemEnumerableIterator`1.MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.RuntimeHelpers.takeInner@667[T,TResult](ConcatEnumerator`2 x, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
    [05:21:04][Step 1/1]at Fake.Globbing.buildPaths(FSharpList`1 acc, FSharpList`1 input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 47
    [05:21:04][Step 1/1]at Fake.Globbing.search(String baseDir, String input) in C:\code\fake\src\app\FakeLib\Globbing\Globbing.fs:line 81
    [05:21:04][Step 1/1]at Fake.FileSystem.System-Collections-Generic-IEnumerable-1-GetEnumerator@39-2.GenerateNext(IEnumerable`1& next) in C:\code\fake\src\app\FakeLib\Globbing\FileSystem.fs:line 40
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.next@185[T](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.next@185[T](FSharpFunc`2 f, IEnumerator`1 e, FSharpRef`1 started, Unit unitVar0)
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.IEnumerator.filter@180.System-Collections-IEnumerator-MoveNext()
    [05:21:04][Step 1/1]at Microsoft.FSharp.Collections.SeqModule.Iterate[T](FSharpFunc`2 action, IEnumerable`1 source)
    [05:21:04][Step 1/1]at Fake.FileHelper.DeleteDir(String path) in C:\code\fake\src\app\FakeLib\FileHelper.fs:line 46
    [05:21:04][Step 1/1]at FSI_0001.Build.del@170-1.Invoke(b _arg2) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 170
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at Fake.TaskRunnerHelper.runWithRetries[a](FSharpFunc`2 f, Int32 retries) in C:\code\fake\src\app\FakeLib\TaskRunnerHelper.fs:line 32
    [05:21:04][Step 1/1]at FSI_0001.Build.removeDir@168(String dir) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 172
    [05:21:04][Step 1/1]at FSI_0001.Build.createNugetPackages[a](a _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 227
    [05:21:04][Step 1/1]at FSI_0001.Build.clo@273-15.Invoke(Unit _arg1) in D:\BuildAgent\work\29e799f160213a61\build.fsx:line 274
    [05:21:04][Step 1/1]at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in C:\code\fake\src\app\FakeLib\TargetHelper.fs:line 411
    [05:21:04][Step 1/1] Process exited with code 42
    [05:21:04][Step 1/1] Step FAKE Build (Command Line) failed
    [05:21:04]Publishing internal artifacts

Ran into this issue with Helios too. Appears to be a regression in a new version of FAKE.

**Comment by Aaronontheweb** [comment]
I think this line is the culprit: https://github.com/akkadotnet/Akka.Persistence.PostgreSql/blob/master/build.fsx#L233

Ran into problems with this with Helios too and ultimately had to hack around it :\, but this code runs fine as-is for Akka.NET core.

**Comment by alexpantyukhin** [comment]
I know It's a little bit late) But maybe it could be useful.

I had the same issue in another project and found that the problem happend when I used together CopyFiles and DeleteDir (in removeDir func).
I rewrote it with CopyDir func and that resolved my problem https://github.com/rogeralsing/Microphone/blob/dotnetcore/build.fsx#L149.
But actualy it's veru strange. Need to review Fake helpers why it happens.
